File: Programming Basics/C++/While loops/Vacation/Vacation.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    double necessaryAmount, savings;
    cin >> necessaryAmount >> savings;

    int consecutiveSpendOperations = 0, totalDays = 0;
    bool canSave = true;
    while (savings < necessaryAmount) {
        string operation;
        double operationAmount;
        cin >> operation >> operationAmount;

        totalDays++;

        if (operation == "save") {
            consecutiveSpendOperations = 0;
            savings += operationAmount;
        }
        else if (operation == "spend") {
            if (++consecutiveSpendOperations == 5) {
                canSave = false;
                break;
            }

            savings = max(savings - operationAmount, 0.0);
        }
    }

    if (canSave) cout << "You saved the money for " << totalDays << " days." << endl;
    else cout << "You can't save the money." << endl << totalDays << endl;
}

File: Programming Basics/JavaScript/While loops/Labs/Exercises/vacation.js
function solve(input) {
    const necessaryAmount = Number(input[0]);
    let savings = Number(input[1]);

    let consecutiveSpendOperations = 0;
    let totalDays = 0;
    let canSave = true;
    let index = 2;
    while (savings < necessaryAmount) {
        const operation = input[index];
        const operationAmount = Number(input[index + 1]);

        totalDays++;

        if (operation === "save") {
            consecutiveSpendOperations = 0;
            savings += operationAmount;
        } else if (operation === "spend") {
            if (++consecutiveSpendOperations == 5) {
                canSave = false;
                break;
            }

            savings = Math.max(savings - operationAmount, 0);
        }

        index += 2;
    }

    if (canSave) console.log(`You saved the money for ${totalDays} days.`);
    else {
        console.log("You can't save the money.");
        console.log(totalDays);
    }
}
